activejob/lib/active_job/test_helper.rb

```diff
@@ -678,7 +678,15 @@ def prepare_args_for_assertion(args)
       def round_time_arguments(argument)
         case argument
         when Time, ActiveSupport::TimeWithZone, DateTime
-          argument.change(usec: 0)
+          nsec =
+            if ActiveJob::Serializers.time_precision > 0
+              nsec = argument.nsec
+              nsec - (nsec % (10 ** (9 - ActiveJob::Serializers.time_precision)))
+            else
+              0
+            end
+
+          argument.change(nsec: nsec)
```

Comment:
So this could be replace with the following:

``` ruby
argument.change(nsec: argument.sec_fraction.floor(ActiveJob::Serializers.time_precision) * 1000000000)
```

Which got me thinking that may be we should add `floor` method to `Time` 🤔 

``` ruby
class Time
  def floor(precision = 0)
    change(nsec: sec_fraction.floor(precision) * 1000000000)
  end
end
```

In which case that code would now be

``` ruby
argument.floor(ActiveJob::Serializers.time_precision)
```

Another alternative would be to add `:sec_fraction` to the list of keys to `change`, e.g:

``` ruby
argument.change(sec_fraction: argument.sec_fraction.floor(ActiveJob::Serializers.time_precision))
```

this would be fairly easy since we calculate the second fraction internal from `:nsec` or `:usec`. In fact we could make both changes to simplify the `floor` method to just this:

``` ruby
class Time
  def floor(precision = 0)
    change(sec_fraction: sec_fraction.floor(precision))
  end
end
```

Can you do a separate PR to add these and we can have a discussion about them - seems like they'd be useful additions. If they're added then we can update this PR to use them.